src/internet/radiobrowser/radiobrowserservice.cpp

```diff
@@ -419,9 +420,28 @@ void RadioBrowserService::ResolveStationUrl(const QUrl& original_url) {
     reply->deleteLater();
     QJsonDocument document = ParseJsonReply(reply);
 
-    QJsonObject item = document.object();
+    StreamList list;
+    QJsonArray contents = document.array();
+    QJsonValue item = contents.first();
+    if (item.isUndefined()) return;
+
+    Song ret;
+    ret.set_valid(true);
+    ret.set_title(item["name"].toString());
     QUrl url(item["url"].toString());
-    emit StationUrlResolved(original_url, url);
+    ret.set_url(url);
+    ret.set_art_automatic(item["favicon"].toString());
+
+    emit StreamMetadataFound(original_url, ret);
+    emit StationUrlResolved(original_url, ret.url());
+
+    // inform the server that we played the station
+    QString determinedUrl =
+        PlayClickUrl.arg(main_server_url_, original_url.path());
+    QUrl playUrl(determinedUrl);
+    QNetworkReply* playReply = network_->get(QNetworkRequest(playUrl));
+    connect(playReply, &QNetworkReply::finished,
+            [=] { playReply->deleteLater(); });
```

Comment:
This last part is where you might consider using the Player::SongChangeRequestProcessed signal.

Reply:
Done, thank you